src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2813,6 +2814,30 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - Moves the tab to another index in the tabs row (if required).
+    // Arguments:
+    // - currentTabIndex: the current index of the tab to move
+    // - suggestedNewTabIndex: the new index of the tab, might get clamped to fit int the tabs row boundaries
+    // Return Value:
+    // - <none>
+    void TerminalPage::_TryMoveTab(const uint32_t currentTabIndex, const int32_t suggestedNewTabIndex)
+    {
+        uint32_t newTabIndex = std::clamp<int32_t>(suggestedNewTabIndex, 0, _tabs.Size() - 1);
+        if (currentTabIndex != newTabIndex)
+        {
+            auto tab = _tabs.GetAt(currentTabIndex);
+            auto tabViewItem = tab.TabViewItem();
+            _tabs.RemoveAt(currentTabIndex);
+            _tabs.InsertAt(newTabIndex, tab);
```

Comment:
It took me a moment to figure out why this is okay. Usually, removing something from a list and then adding with a precalculated index that hasn't been adjusted for the removal is frightening 😁

Reply:
But it's fine because we're not trying to hit the _same_ place we just vacated, we're trying to hit the next one.

Reply:
Oh.. it took me a while getting to it.. 😄 I wrote entire function.. read it 10 times to understand if it is correct.. and instead understood it is obsolete 🤦